### PR TITLE
Make log audit backend configurable in GCE

### PR DESCRIPTION
**Pull request by crassirostris**
This PR will allow to enable audit logging batching by default in e2e tests, after https://github.com/kubernetes/kubernetes/pull/60739 is merged. This is an important step to prevent a regression in scale tests.

/cc @tallclair @sttts 

/assign @roberthbailey 

Robert, please approve

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/config-test.sh [hunk 1]
```diff
@@ -377,6 +377,7 @@ ENABLE_LEGACY_ABAC="${ENABLE_LEGACY_ABAC:-false}" # true, false
 
 # Enable a simple "AdvancedAuditing" setup for testing.
 ENABLE_APISERVER_ADVANCED_AUDIT="${ENABLE_APISERVER_ADVANCED_AUDIT:-true}" # true, false
+ADVANCED_AUDIT_LOG_MODE="${ADVANCED_AUDIT_LOG_MODE:-batch}" # batch, blocking
```
Comment (roberthbailey) [inline cluster/gce/config-test.sh]: Do you need to set this in config-default.sh too?

Comment (tallclair) [reply cluster/gce/config-test.sh]: I think I'd prefer to leave the non-test default as blocking until we fix an issue that could lead to an out-of-order log.

**Comment by tallclair** [comment]
Follow up:
Change some test suites to overide the mode to blocking, so we have coverage of both.

**Pull request merged**